packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettings.kt

```diff
@@ -26,23 +28,46 @@ class PluginSettingsStateComponent : SimplePersistentStateComponent<PluginSettin
 
 /**
  * The settings themselves. They are tracked, so any change on the settings properties will be eventually
- * persisted by IntelliJ. To access the settings, use the useSettings provider.
+ * persisted by IntelliJ. To access the settings, use the pluginSetting provider. For example,
+ * let's say you want to get the isTelemetryEnabled setting, you would do:
  *
- * @see useSettings
+ * ```kt
+ * val isTelemetryEnabled by pluginSetting { ::isTelemetryEnabled }
+ * ```
+ *
+ * You'll see the settings as the underlying type (in this case, a boolean). If you want to modify
+ * the setting, specify a variable instead of a value:
+ *
+ * ```kt
+ * var isTelemetryEnabled by pluginSetting { ::isTelemetryEnabled }
+ * isTelemetryEnabled = false
+ * ```
+ *
+ * @see pluginSetting
  */
 class PluginSettings : BaseState(), Serializable {
     var isTelemetryEnabled by property(true)
     var hasTelemetryOptOutputNotificationBeenShown by property(false)
 }
 
-/**
- * Function that provides a reference to the current PluginSettings.
- *
- * @see PluginSettings
- *
- * @return
- */
-fun useSettings(): PluginSettings =
-    ApplicationManager.getApplication().getService(
-        PluginSettingsStateComponent::class.java,
-    ).state
+class SettingsDelegate<T>(private val settingProp: KMutableProperty0<T>) {
+    operator fun getValue(
+        thisRef: Any?,
+        property: KProperty<*>
+    ): T {
+        return settingProp.getter.call()
+    }
+
+    operator fun setValue(
+        thisRef: Any?,
+        property: KProperty<*>,
+        value: T
+    ) {
+        settingProp.setter.call(value)
+    }
+}
+
+fun <T> pluginSetting(cb: PluginSettings.() -> KMutableProperty0<T>): SettingsDelegate<T> {
+    val settings by service<PluginSettingsStateComponent>()
+    return SettingsDelegate(settings.state.cb())
+}
```

Comment:
This is really amazing. I will check Kotlin docs for more reference to this, feel free to drop any other reference that might be helpful.